src/login/login.component.tsx

```diff
@@ -5,18 +5,33 @@ import { performLogin } from "./login.resource";
 export default function Login(props: LoginProps) {
   const [username, setUsername] = React.useState("");
   const [password, setPassword] = React.useState("");
+  const [authenticated, setAuthenticated] = React.useState(null);
+  const [errorMessage, setErrorMessage] = React.useState("");
   const [isLoggingIn, setIsLoggingIn] = React.useState(false);
   const [showPassword, setShowPassword] = React.useState(false);
   const passwordInputRef = React.useRef<HTMLInputElement>(null);
   const usernameInputRef = React.useRef<HTMLInputElement>(null);
+  const formRef = React.useRef<HTMLFormElement>(null);
 
   React.useEffect(() => {
     if (isLoggingIn) {
       performLogin(username, password)
         .then(data => {
-          props.history.push("/patient-search");
+          // workaround for now, @openmrs/esm-api response.json() not working!
+          const authData = data["data"];
+          if (authData) {
+            const { authenticated } = authData;
+            setAuthenticated(authenticated);
```

Comment:
What's the point of setting state right before we navigate away?

Reply:
you're right, I should probably move this down to the else statement